### PR TITLE
feat: subscription

### PR DESCRIPTION
**Pull request by CristhianF7**

### CODE REVIEW

File: app/lib/common.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    });
+
+    return result?.data;
+  } catch (error) {
+    // supressing error. license not found
+    return {} as License;
+  }
+}
+
+export async function getFeatureFlas() {
```
Comment (johndietz) [inline app/lib/common.ts:21]: not a blocker, just calling out an apparent typo on function name

**Review by johndietz — APPROVED** [review]

File: app/lib/common.ts (hunk 1 above)
Comment (CristhianF7) [reply app/lib/common.ts:21]: Thanks, great catch!!

File: containers/license/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          <>
+            <Typography
+              variant="subtitle2"
+              color={VOLCANIC_SAND}
+            >{`You are on a ${license?.plan.name} Plan`}</Typography>
+            <Typography variant="labelLarge" color={EXCLUSIVE_PLUM} sx={{ mt: 1, mb: 1 }}>
+              License key
+            </Typography>
+            <Typography variant="body2" color={VOLCANIC_SAND}>
+              {`${license?.licenseKey
```
Comment (D-B-Hawk) [inline containers/license/index.tsx:60]:
```suggestion
             export function formatLicenseKey(licenseKey:string){
                return licenseKey.slice(0, 4)
                .toUpperCase()}************************************`}
             }
```

File: redux/thunks/subscription.thunk.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+export const activateLicenseKey = createAsyncThunk<
+  License,
+  string,
+  {
+    state: RootState;
+  }
+>('subscription/activateLicenseKey', async (licenseKey, { getState }) => {
+  const { managementCluster } = getState().api;
+
+  const result = await axios.post<License>('/api/proxy?target=ee', {
```
Comment (D-B-Hawk) [inline redux/thunks/subscription.thunk.ts:32]: same suggestion as noted above

File: app/settings/subscription/page.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const response = await axios.get(`${SAAS_API_URL}/api/v1/payment/plans`);
+
+  return sortBy(
+    response.data,
+    (product) => product && product.metadata && product.metadata['order'],
+  );
+}
+
+export default async function Page() {
+  const plans = await getProductPlans();
```
[the remark this answers is not in the document]
Comment (D-B-Hawk) [reply app/settings/subscription/page.tsx:17]: ok. well in that case there should probably be some error handling if these calls fail.

File: app/dashboard/layout.tsx [hunk 5]
```diff
@@ -1,45 +1,17 @@
...
-      setLoadFlags(true);
-      dispatch(getFlags());
-      dispatch(getEnvVariables());
-      dispatch(getClusters());
-    }
-  }, [dispatch, loadFlags]);
+export default async function Page({ children }: PropsWithChildren) {
+  const license = await validateLicense();
+  const envVariables = await getEnvVars();
+  const featureFlags = await getFeatureFlags();
```
Comment (D-B-Hawk) [inline app/dashboard/layout.tsx:10]: these calls will need error handling

File: app/lib/common.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    return result?.data;
+  } catch (error) {
+    // supressing error. license not found
+    return {} as License;
+  }
+}
+
+export async function getFeatureFlags() {
+  const result = await axios.get<{ flags: Record<FeatureFlag, boolean> }>(`${BASE_URL}/api/flags`);
```
Comment (D-B-Hawk) [inline app/lib/common.ts:22]: wrap in a try catch as above in validate license

**Review by D-B-Hawk — APPROVED** [review]
🔥 🚀

**Pull request merged**